pootle/apps/pootle_translationproject/models.py

```diff
@@ -293,21 +286,7 @@ def save(self, *args, **kwargs):
                 for template_store in template_stores.iterator():
                     init_store_from_template(self, template_store)
 
-            self.scan_files()
-
-            # If this TP has no stores, cache should be updated forcibly.
-            if self.stores.live().count() == 0:
-                self.update_all_cache()
-
-            # Create units from disk store
-            for store in self.stores.live().iterator():
-                changed = store.update_from_disk()
-
-                # If there were changes stats will be refreshed anyway -
-                # otherwise...  Trigger stats refresh for TP added from UI.
-                # FIXME: This won't be necessary once #3547 is fixed.
-                if not changed:
-                    store.save(update_cache=True)
+                self.update_from_disk()
```

Comment:
@unho there was my intention to avoid having `TP.update_from_disk()` in `TP.save()`. I put it explicitly to the [signal receivers]( https://github.com/ta2-1/pootle/blob/tp_cleanups/pootle/apps/pootle_translationproject/models.py#L481-L492).